### PR TITLE
test: update to jest 29

### PR DESCRIPTION
**Pull request by rodleyorosa**

### CODE REVIEW

File: src/components/basic/Button.test.tsx [hunk 1]
```diff
@@ -7,93 +7,81 @@
...
 	test('Show tooltip on disabled button', async () => {
 		const clickFn = jest.fn();
-		render(
+		const { user } = setup(
 			<Tooltip label={'Tooltip label'}>
 				<Button label={'Button'} loading onClick={clickFn} disabled />
 			</Tooltip>
 		);
 		// FIXME: hover event on disabled button is not bubbled up. Remove access on parentElement when possible
 		// eslint-disable-next-line testing-library/no-node-access
```
Comment (beawar) [inline src/components/basic/Button.test.tsx]: both comment lines can be removed since it is now fixed and no parentElement is accessed anymore

File: src/components/display/CollapsingActions.test.tsx [hunk 2]
```diff
@@ -91,7 +90,7 @@ describe('Collapsing Actions', () => {
 	test('Resize event makes action to be collapsed or visible based on width', async () => {
 		/*
 		 * Considering that icon has 32px of width by default, 10 actions require 320px to be all visible.
-		 * Default width of the window in jest is 1024, so the container is rendered initially with a
+		 * Default width of the window in jest is 1024, so the container is setuped initially with a
```
Comment (beawar) [inline src/components/display/CollapsingActions.test.tsx]: I think rendered is fine :D Find and replace went wrong here XD

File: src/components/display/Dropdown.test.tsx [hunk 3]
```diff
@@ -72,17 +71,17 @@ describe('Dropdown', () => {
...
+		await user.click(screen.getByRole('button'));
 
-		expect(screen.getByText('Some Item')).toBeInTheDocument();
-		expect(screen.getByText('Some Other Item')).toBeInTheDocument();
-		expect(screen.getByText('Yet Another Item')).toBeInTheDocument();
+		expect(screen.getByText('Some Item')).toBeVisible();
+		expect(screen.getByText('Some Other Item')).toBeVisible();
+		expect(screen.getByText('Yet Another Item')).toBeVisible();
 		// wait for listeners to be registered
 		await waitFor(
```
Comment (beawar) [inline src/components/display/Dropdown.test.tsx]: replace all these waitFor with advanceTimersByTime

File: src/components/display/Dropdown.test.tsx [hunk 4]
```diff
@@ -144,21 +143,23 @@ describe('Dropdown', () => {
...
 			<Modal open title="modal with dropdown" onClose={onClose}>
 				<Dropdown items={items}>
 					<Button label="opener" onClick={onClick} />
 				</Dropdown>
 			</Modal>
 		);
-
+		// run timers of modal
+		act(() => jest.runOnlyPendingTimers());
+		jest.runOnlyPendingTimers();
```
Comment (beawar) [inline src/components/display/Dropdown.test.tsx]: I think we can replace one of the two with a `await screen.findByText('modal with dropdown')`. Then, after the element has been found (it is in the document), we can advance timers to make the element be visible, with something like `jest.advanceTimersByTime(TIMERS.modal.delayOpen)` (an act could be required)

File: src/components/display/Dropdown.test.tsx [hunk 5]
```diff
@@ -291,69 +292,45 @@ describe('Dropdown', () => {
...
 		expect(screen.getByText('item 3')).toBeVisible();
 		expect(screen.queryByText(/tooltip/i)).not.toBeInTheDocument();
-		userEvent.hover(screen.getByText('item 1'));
-		await waitFor(
-			() =>
-				new Promise((resolve) => {
-					setTimeout(resolve, 500);
-				})
-		);
+		await user.hover(screen.getByText('item 1'));
```
Comment (beawar) [inline src/components/display/Dropdown.test.tsx:259]: Advance timers here to be sure that the tooltip is not shown. The timeout is 500, which is the default timeout of the tooltip (use a const to save this timer inside TIMER const)

File: src/components/display/Dropdown.test.tsx [hunk 6]
```diff
@@ -291,69 +292,45 @@ describe('Dropdown', () => {
...
+			await user.unhover(screen.getByText('item 2'));
 		});
-		userEvent.hover(screen.getByText('item 3'));
-		await waitFor(
-			() =>
-				new Promise((resolve) => {
-					setTimeout(resolve, 500);
-				})
-		);
+		await user.hover(screen.getByText('item 3'));
```
Comment (beawar) [inline src/components/display/Dropdown.test.tsx:268]: Also here the 500 was placed to be sure to wait for tooltip default delay

File: src/components/display/Table.test.tsx [hunk 7]
```diff
@@ -92,7 +94,7 @@ describe('Table', () => {
 		);
 		const row1 = getRowByColumnLabel('row1col1');
 		const row1Checkbox = within(row1).getByTestId(SELECTORS.checkbox);
-		expect(row1Checkbox).toBeInTheDocument();
+		expect(row1Checkbox).toBeVisible();
 		expect(row1Checkbox).toHaveStyleRule('display', 'none');
```
Comment (beawar) [inline src/components/display/Table.test.tsx]: in this test we are checking actually that the element is in the document but not visible (display css property set to none). I think we could update the test to be
```suggestion
		expect(row1Checkbox).toBeInTheDocument();
		expect(row1Checkbox).not.toBeVisible();
```

File: src/components/display/Tooltip.test.tsx [hunk 8]
```diff
@@ -48,20 +46,15 @@ describe('Tooltip', () => {
...
 		// wait so tooltip can register the listeners
-		await new Promise((r) => {
-			setTimeout(r, 100);
-		});
-		userEvent.hover(button);
-		await new Promise((r) => {
-			setTimeout(r, 100);
-		});
+		jest.advanceTimersByTime(TIMERS.TOOLTIP.REGISTER_LISTENER);
+		await user.hover(button);
```
Comment (beawar) [inline src/components/display/Tooltip.test.tsx:50]: advance timers of tooltip default delay to be sure the tooltip is not displayed

File: src/components/feedback/Banner.test.tsx [hunk 9]
```diff
@@ -61,8 +60,8 @@ describe('Banner', () => {
 		});
 	}
 
-	async function openMoreInfoModal(): Promise<HTMLElement> {
-		userEvent.click(screen.getByRole('button', { name: /more info/i }));
+	async function openMoreInfoModal(user: UserEvent): Promise<HTMLElement> {
+		await user.click(screen.getByRole('button', { name: /more info/i }));
 		const modal = await screen.findByTestId('modal');
 
 		// run modal timeout
```
Comment (beawar) [inline src/components/feedback/Banner.test.tsx:68]: replace the following waitFor with a jest.advanceTimersByTime(TIMERS.MODAL.DELAY_OPEN)

File: src/components/inputs/DateTimePicker.test.tsx [hunk 10]
```diff
@@ -187,48 +186,53 @@ describe('DateTimePicker', () => {
...
+			const { user } = setup(<DateTimePicker label={'Validate input'} enableChips />);
 			const inputElement = screen.getByRole('textbox');
 			expect(inputElement).toBeEnabled();
-			userEvent.click(inputElement);
-			userEvent.click(screen.getAllByText('1')[0]);
+			await user.click(inputElement);
+			await user.click(screen.getAllByText('1')[0]);
+			act(() => {
+				jest.runOnlyPendingTimers();
+			});
```
Comment (beawar) [inline src/components/inputs/DateTimePicker.test.tsx]: why is this needed?

File: src/components/inputs/MultiButton.test.tsx [hunk 11]
```diff
@@ -59,16 +58,16 @@ describe('MultiButton', () => {
...
 		];
 		const clickFn = jest.fn();
-		render(<MultiButton items={items} onClick={clickFn} label="primary" />);
+		const { user } = setup(<MultiButton items={items} onClick={clickFn} label="primary" />);
 
 		expect(screen.getByText(/primary/i)).toBeVisible();
 		expect(screen.getByTestId(ICONS.multiButtonSecondaryAction)).toBeVisible();
-		userEvent.click(screen.getByTestId(ICONS.multiButtonSecondaryAction));
+		await user.click(screen.getByTestId(ICONS.multiButtonSecondaryAction));
 		await waitForListenerToBeRegistered(2);
```
Comment (beawar) [inline src/components/inputs/MultiButton.test.tsx]: remove this util or replace its implementation to use jest.advanceTimersByTime

File: src/components/inputs/Radio.test.tsx [hunk 12]
```diff
@@ -33,132 +32,136 @@ describe('Radio', () => {
...
 
 	describe('Uncontrolled mode', () => {
-		test('should call onChange with the new checked status', () => {
+		test('should call onChange with the new checked status', async () => {
 			const onChangeFn = jest.fn();
-			render(<Radio label={'the label'} onChange={onChangeFn} defaultChecked />);
+			const { user } = setup(<Radio label={'the label'} onChange={onChangeFn} defaultChecked />);
 			const radio = screen.getByRole('radio');
-			// onChange is called on render
+			// onChange is called on setup
```
Comment (beawar) [inline src/components/inputs/Radio.test.tsx]: keep `render` here, it is meaningful
```suggestion
			// onChange is called on render
```

File: src/components/inputs/Radio.test.tsx [hunk 13]
```diff
@@ -33,132 +32,136 @@ describe('Radio', () => {
...
 		});
 	});
 
 	describe('Controlled mode', () => {
 		test('should call onChange with the new checked status when checked prop change', () => {
 			const onChangeFn = jest.fn();
-			const { rerender } = render(<Radio label={'the label'} onChange={onChangeFn} checked />);
-			// onChange is called on render
+			const { rerender } = setup(<Radio label={'the label'} onChange={onChangeFn} checked />);
+			// onChange is called on setup
```
Comment (beawar) [inline src/components/inputs/Radio.test.tsx]:
```suggestion
			// onChange is called on render
```

File: src/components/inputs/Radio.test.tsx [hunk 14]
```diff
@@ -33,132 +32,136 @@ describe('Radio', () => {
...
 
-		test('should not toggle the radio on click', () => {
+		test('should not toggle the radio on click', async () => {
 			const onChangeFn = jest.fn();
-			render(<Radio label={'the label'} checked onChange={onChangeFn} />);
+			const { user } = setup(<Radio label={'the label'} checked onChange={onChangeFn} />);
 			const radio = screen.getByRole('radio');
 			expect(radio).toBeChecked();
-			// onChange is called on render
+			// onChange is called on setup
```
Comment (beawar) [inline src/components/inputs/Radio.test.tsx]:
```suggestion
			// onChange is called on render
```

File: src/components/inputs/RadioGroup.test.tsx [hunk 15]
```diff
@@ -6,16 +6,15 @@
...
 
 import { Radio } from './Radio';
 import { RadioGroup, RadioGroupProps } from './RadioGroup';
-import { render } from '../../test-utils';
+import { setup } from '../../test-utils';
 
 describe('Radio Group', () => {
-	test('Should render a group with the radio', () => {
-		render(
+	test('Should setup a group with the radio', () => {
```
Comment (beawar) [inline src/components/inputs/RadioGroup.test.tsx]:
```suggestion
	test('Should render a group with the radio', () => {
```

File: src/components/inputs/Slider.test.tsx [hunk 16]
```diff
@@ -5,57 +5,54 @@
...
 		const options = ['opt1', 'opt2', 'opt3', 'opt4', 'opt5'];
-		render(<Slider options={options} />);
+		const { user } = setup(<Slider options={options} />);
 		const option1 = screen.getByRole('option', { name: 'opt1' });
 		expect(option1).toBeVisible();
-		// wait so tooltip can register the listeners
-		await new Promise((r) => {
-			setTimeout(r, 100);
+		act(() => {
+			jest.runOnlyPendingTimers();
```
Comment (beawar) [inline src/components/inputs/Slider.test.tsx]: use advanceTimersByTime(TIMERS.TOOLTIP.REGISTER_LISTENER)

File: src/test-utils.tsx [hunk 17]
```diff
@@ -68,15 +79,36 @@ const customQueries = {
...
+);
 
-	return rtlRender(ui, {
-		wrapper: Wrapper,
+function customRender(
+	ui: React.ReactElement,
+	options: Omit<RenderOptions, 'queries' | 'wrapper'> = {}
+): RenderResult<typeof queriesExtended> {
+	return render(ui, {
+		wrapper: ({ children }: Pick<WrapperProps, 'children'>) => <Wrapper>{children}</Wrapper>,
```
Comment (beawar) [inline src/test-utils.tsx]: we can keep `wrapper: Wrapper` here since we are not passing additional props to the wrapper component

File: src/components/display/Dropdown.test.tsx [hunk 18]
```diff
@@ -144,52 +129,47 @@ describe('Dropdown', () => {
...
+		const { user } = setup(
 			<Modal open title="modal with dropdown" onClose={onClose}>
 				<Dropdown items={items}>
 					<Button label="opener" onClick={onClick} />
 				</Dropdown>
 			</Modal>
 		);
-
+		// run timers of modal
+		act(() => jest.runOnlyPendingTimers());
```
Comment (beawar) [inline src/components/display/Dropdown.test.tsx]: this line should not be needed anymore since we are waiting for the text now with the findBy

File: src/components/display/Dropdown.test.tsx [hunk 19]
```diff
@@ -268,92 +238,67 @@ describe('Dropdown', () => {
...
 		});
-		userEvent.hover(screen.getByText('item 3'));
-		await waitFor(
-			() =>
-				new Promise((resolve) => {
-					setTimeout(resolve, 500);
-				})
-		);
+		// wait so tooltip can register the listeners
+		jest.advanceTimersByTime(TIMERS.TOOLTIP.REGISTER_LISTENER);
```
Comment (beawar) [inline src/components/display/Dropdown.test.tsx]: not sure, but I would say this advance is not needed, the listeners should be registered already

File: src/components/display/Dropdown.test.tsx [hunk 20]
```diff
@@ -268,92 +238,67 @@ describe('Dropdown', () => {
...
+		await user.hover(screen.getByText('item 3'));
```
Comment (beawar) [inline src/components/display/Dropdown.test.tsx:268]: After this hover, we need to advance timers by 500ms (create a constant TIMERS.TOOLTIP.DELAY_SHOW or similar). We need to advance timers to be sure that the tooltip is not shown.

File: src/components/display/Tooltip.test.tsx [hunk 21]
```diff
@@ -48,20 +46,16 @@ describe('Tooltip', () => {
...
+		jest.advanceTimersByTime(TIMERS.TOOLTIP.REGISTER_LISTENER);
```
Comment (beawar) [inline src/components/display/Tooltip.test.tsx]: this advance should use 500ms (TIMERS.TOOLTIP.DELAY_SHOW)

**Review by beawar — DISMISSED** [review]

**Review by lisaparma — DISMISSED** [review]

File: src/components/display/Dropdown.test.tsx [hunk 22]
```diff
@@ -268,92 +236,67 @@ describe('Dropdown', () => {
...
+		await user.hover(screen.getByText('item 1'));
+		jest.advanceTimersByTime(TIMERS.TOOLTIP.DELAY_SHOW);
 		expect(screen.queryByText(/tooltip/i)).not.toBeInTheDocument();
-		userEvent.hover(screen.getByText('item 2'));
+		await user.hover(screen.getByText('item 2'));
 		await screen.findByText(/tooltip/i);
 		expect(screen.getByText('tooltip 2')).toBeVisible();
-		act(() => {
-			userEvent.unhover(screen.getByText('item 2'));
+		await act(async () => {
```
Comment (nubsthead) [inline src/components/display/Dropdown.test.tsx:265]: did you try using await waitFor? should do the same but in a more readable way (not sure tho)

**Review by nubsthead — DISMISSED** [review]

**Review by CataldoMazzilli — APPROVED** [review]

File: src/components/display/Dropdown.test.tsx (hunk 22 above)
Comment (beawar) [reply src/components/display/Dropdown.test.tsx:265]: Have you a link or some doc where it is suggested to use the user with the waitFor?

**Review by beawar — APPROVED** [review]
